crates/solidity/outputs/cargo/crate/src/generated/language.rs

```diff
@@ -5057,36 +4131,38 @@ impl Language {
 
     #[allow(unused_assignments, unused_parens)]
     fn type_name(&self, input: &mut ParserContext<'_>) -> ParserResult {
-        #[allow(unused_variables)]
-        let parse_array_type_name = |input: &mut ParserContext<'_>| {
+        let parse_postfix_array_type_name = |input: &mut ParserContext<'_>| {
             PrecedenceHelper::to_postfix_operator(
                 RuleKind::ArrayTypeName,
                 1u8,
-                SequenceHelper::run(|mut seq| {
-                    let mut delim_guard = input.open_delim(TokenKind::CloseBracket);
-                    let input = delim_guard.ctx();
-                    seq.elem(self.parse_token_with_trivia::<LexicalContextType::Default>(
-                        input,
-                        TokenKind::OpenBracket,
-                    ))?;
-                    seq.elem(
-                        OptionalHelper::transform(self.expression(input))
-                            .recover_until_with_nested_delims::<_, LexicalContextType::Default>(
+                ChoiceHelper::run(input, |mut choice, input| {
```

Comment:
Not sure if the added `ChoiceHelper` here is intended. How does this affect the output?

Reply:
Yes. Thank you.